### PR TITLE
Add a new "corner' layout

### PR DESCRIPTION
**Pull request by AlexisBRENON**
Create a new layout called corner, with different orientation.
See the joined image or created icons to see the results
Example from Thomas/Schnouki

### CODE REVIEW

**Comment by AlexisBRENON** [comment]
Comments updated

**Comment by Elv13** [comment]
Thanks for this

I use this kind of layout in my config, I think it really is the best terminal layout for 24"+ screens.

**Comment by AlexisBRENON** [comment]
How can the Travis CI build fail while I only changed comments?!

**Comment by egordorichev** [comment]
Thank you! Looks nice!

**Comment by blueyed** [comment]
Travis timed out. I have restarted it.

**Comment by Schnouki** [comment]
Travis timed out again, during the `git clone` :/
Otherwise this layout looks really nice, and I really hope this PR will be merged. :+1:

**Comment by blueyed** [comment]
As for Travis, this should be fixed in master now.

As for this PR: @AlexisBRENON can you please squash your commits and rebase it on current master?  I think it's good to be merged then.

**Comment by AlexisBRENON** [comment]
@blueyed Have I to squash all the commits in only one !?

File: awesomerc.lua.in [hunk 1]
```diff
@@ -65,7 +65,11 @@ awful.layout.layouts = {
     awful.layout.suit.spiral.dwindle,
     awful.layout.suit.max,
     awful.layout.suit.max.fullscreen,
-    awful.layout.suit.magnifier
+    awful.layout.suit.magnifier,
+    awful.layout.suit.corner.nw,
+    awful.layout.suit.corner.ne,
+    awful.layout.suit.corner.se,
+    awful.layout.suit.corner.sw
```
Comment (blueyed) [inline awesomerc.lua.in]: I am not sure if all variants should be enabled in the default config?

File: awesomerc.lua.in [hunk 2]
```diff
@@ -261,6 +265,7 @@ globalkeys = awful.util.table.join(
     awful.key({ modkey, "Control" }, "l",     function () awful.tag.incncol(-1)         end),
     awful.key({ modkey,           }, "space", function () awful.layout.inc( 1) end),
     awful.key({ modkey, "Shift"   }, "space", function () awful.layout.inc(-1) end),
+    awful.key({ modkey,           }, "s", function () awful.layout.suit.corner.swap_privileged(); awful.layout.arrange(mouse.screen) end),
```
Comment (blueyed) [inline awesomerc.lua.in]: I am not sure if this binding is useful/unspecific enough to have it by default.
If so, it should be also added to the man page etc.

File: lib/awful/layout/suit/corner.lua.in [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+-- @author Alexis Brenon &lt;brenon.alexis+awesomewm@gmail.com&gt;
+-- @copyright 2015 Alexis Brenon
+-- @release @AWESOME_VERSION@
+
+-- Grab environment we need
+local ipairs = ipairs
+local math = math
+local tag = require("awful.tag")
+
+local function do_corner(p, orientation, row_privileged)
```
Comment (blueyed) [inline lib/awful/layout/suit/corner.lua.in]: Please document the function (arguments).

File: lib/awful/layout/suit/corner.lua.in [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+            }
+            row.win_idx = row.win_idx + 1
+        end
+        c:geometry(g)
+    end
+end
+
+local corner = {}
+corner.row_privileged = false
+--- Switch which one of row or column will contains more slave clients
```
Comment (blueyed) [inline lib/awful/layout/suit/corner.lua.in]: s/contains/contain/

**Comment by blueyed** [comment]
> @blueyed Have I to squash all the commits in only one !?

I'd do so: it's a new feature, and it does not make sense to have separate commits for adding it to each default theme for example.

File: awesomerc.lua.in (hunk 2 above)
Comment (blueyed) [inline awesomerc.lua.in]: Maybe this could be controlled using `nmaster` (which does not appear to be used otherwise in this layout)?

**Comment by AlexisBRENON** [comment]
Rework done:
Remove the swap_privileged function and use the nmaster value instead
Add comments to the function (it's not displayed in documentation because function is not exported)
Use only one variant in the default rc.lua

Two commits because of the merge commit. Is it good for you or do you prefer only one commit (doing a rebase) ?

Cheers.

**Comment by blueyed** [comment]
Thanks!

> Use only one variant in the default rc.lua

I will re-add the other 3 variants, but commented.

> Two commits because of the merge commit. Is it good for you or do you prefer only one commit (doing a rebase) ?

A single commit would be better - you should have probably created the PR using your 'feature/corner_layout' branch.  But it's fine like that now.

**Pull request closed without merging**